Not GHC 8.0.1 compatible

Building this package fails because of some ambiguity introduced by ghc-8.0.1. For example:

```
 /private/var/folders/xp/8fbnz73x77786vmh_8zhct3w0000gn/T/stack38582/protocol-buffers-2.3.1/Text/ProtocolBuffers/Get.hs:211:50: error:
        • Could not deduce (Integral a7)
            arising from a use of ‘fromIntegral’
          from the context: (Num s, Integral s, Bits s)
            bound by the type signature for:
                       decode7unrolled :: (Num s, Integral s, Bits s) => Get s
            at Text/ProtocolBuffers/Get.hs:133:1-64
          The type variable ‘a7’ is ambiguous
          Relevant bindings include
            b'A :: a7 (bound at Text/ProtocolBuffers/Get.hs:210:17)
          These potential instances exist:
            instance Integral Int32 -- Defined in ‘GHC.Int’
            instance Integral Int64 -- Defined in ‘GHC.Int’
            instance Integral Word16 -- Defined in ‘GHC.Word’
            ...plus six others
            ...plus six instances involving out-of-scope types
            (use -fprint-potential-instances to see them all)
        • In the first argument of ‘shiftL’, namely ‘fromIntegral b'A’
          In the second argument of ‘(.|.)’, namely
            ‘(fromIntegral b'A `shiftL` 63)’
          In the first argument of ‘ok’, namely
            ‘(val'9 .|. (fromIntegral b'A `shiftL` 63))’
```

Full log: https://gist.github.com/MaxDaten/9fbd4213bbfc168fcc90917de1a11cba

I'm working on a PR.

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,6 +6,7 @@ before_install:
   - mkdir -p ~/.local/bin
   - export PATH=~/.local/bin:$PATH
   - curl -L https://github.com/commercialhaskell/stack/releases/download/v0.1.4.0/stack-0.1.4.0-x86_64-linux.tar.gz | tar xz -C ~/.local/bin
+  - stack --no-terminal --skip-ghc-check install happy alex
 
 matrix:
   include:

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,7 @@ matrix:
       addons: {apt: {packages: [ghc-7.8.4], sources: [hvr-ghc]}}
     - env: GHCVER=7.10.2 STACK_YAML=stack-7.10.yaml
       addons: {apt: {packages: [ghc-7.10.2],sources: [hvr-ghc]}}
-    - env: GHCVER=7.10.2 STACK_YAML=stack-8.0.1.yaml
+    - env: GHCVER=8.0.1 STACK_YAML=stack-8.0.1.yaml
       addons: {apt: {packages: [ghc-8.0.1],sources: [hvr-ghc]}}
     - env: GHCVER=head STACK_YAML=stack-head.yaml
       addons: {apt: {packages: [cabal-install-head,ghc-head],  sources: [hvr-ghc]}}
@@ -26,8 +26,8 @@ matrix:
    - env: GHCVER=head STACK_YAML=stack-head
 
 install:
-  - stack --no-terminal --skip-ghc-check install happy alex
   - stack --no-terminal --skip-ghc-check setup
+  - stack --no-terminal --skip-ghc-check install happy alex
   - stack --no-terminal --skip-ghc-check test --only-snapshot
 
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,6 +13,8 @@ matrix:
       addons: {apt: {packages: [ghc-7.8.4], sources: [hvr-ghc]}}
     - env: GHCVER=7.10.2 STACK_YAML=stack-7.10.yaml
       addons: {apt: {packages: [ghc-7.10.2],sources: [hvr-ghc]}}
+    - env: GHCVER=7.10.2 STACK_YAML=stack-8.0.1.yaml
+      addons: {apt: {packages: [ghc-8.0.1],sources: [hvr-ghc]}}
     - env: GHCVER=head STACK_YAML=stack-head.yaml
       addons: {apt: {packages: [cabal-install-head,ghc-head],  sources: [hvr-ghc]}}
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,15 @@
 language: c
 sudo: false
 
+cache:
+  directories:
+  - $HOME/.stack
+
 before_install:
   # stack
   - mkdir -p ~/.local/bin
   - export PATH=~/.local/bin:$PATH
   - curl -L https://github.com/commercialhaskell/stack/releases/download/v0.1.4.0/stack-0.1.4.0-x86_64-linux.tar.gz | tar xz -C ~/.local/bin
-  - stack --no-terminal --skip-ghc-check install happy alex
 
 matrix:
   include:
@@ -23,6 +26,7 @@ matrix:
    - env: GHCVER=head STACK_YAML=stack-head
 
 install:
+  - stack --no-terminal --skip-ghc-check install happy alex
   - stack --no-terminal --skip-ghc-check setup
   - stack --no-terminal --skip-ghc-check test --only-snapshot
 

--- a/Text/ProtocolBuffers/Get.hs
+++ b/Text/ProtocolBuffers/Get.hs
@@ -165,49 +165,49 @@ decode7unrolled = Get $ \ sc sIn@(S ss@(S.PS fp off len) bs n) pc -> trace ("dec
                     !ptr3 = ptr2 `plusPtr` 1
                 if ptr3 >= end then more else do
 
-                b'3 <- peek ptr3
+                b'3::Word8 <- peek ptr3
                 if b'3 < 128 then ok (val'2 .|. (fromIntegral b'3 `shiftL` 14)) 3 else do
                 let !val'3 = (val'2 .|. (fromIntegral (b'3 .&. 0x7F) `shiftL` 14))
                     !ptr4 = ptr3 `plusPtr` 1
                 if ptr4 >= end then more else do
 
-                b'4 <- peek ptr4
+                b'4::Word8 <- peek ptr4
                 if b'4 < 128 then ok (val'3 .|. (fromIntegral b'4 `shiftL` 21)) 4 else do
                 let !val'4 = (val'3 .|. (fromIntegral (b'4 .&. 0x7F) `shiftL` 21))
                     !ptr5 = ptr4 `plusPtr` 1
                 if ptr5 >= end then more else do
 
-                b'5 <- peek ptr5
+                b'5::Word8 <- peek ptr5
                 if b'5 < 128 then ok (val'4 .|. (fromIntegral b'5 `shiftL` 28)) 5 else do
                 let !val'5 = (val'4 .|. (fromIntegral (b'5 .&. 0x7F) `shiftL` 28))
                     !ptr6 = ptr5 `plusPtr` 1
                 if ptr6 >= end then more else do
 
-                b'6 <- peek ptr6
+                b'6::Word8 <- peek ptr6
                 if b'6 < 128 then ok (val'5 .|. (fromIntegral b'6 `shiftL` 35)) 6 else do
                 let !val'6 = (val'5 .|. (fromIntegral (b'6 .&. 0x7F) `shiftL` 35))
                     !ptr7 = ptr6 `plusPtr` 1
                 if ptr7 >= end then more else do
 
-                b'7 <- peek ptr7
+                b'7::Word8 <- peek ptr7
                 if b'7 < 128 then ok (val'6 .|. (fromIntegral b'7 `shiftL` 42)) 7 else do
                 let !val'7 = (val'6 .|. (fromIntegral (b'7 .&. 0x7F) `shiftL` 42))
                     !ptr8 = ptr7 `plusPtr` 1
                 if ptr8 >= end then more else do
 
-                b'8 <- peek ptr8
+                b'8::Word8 <- peek ptr8
                 if b'8 < 128 then ok (val'7 .|. (fromIntegral b'8 `shiftL` 49)) 8 else do
                 let !val'8 = (val'7 .|. (fromIntegral (b'8 .&. 0x7F) `shiftL` 49))
                     !ptr9 = ptr8 `plusPtr` 1
                 if ptr9 >= end then more else do
 
-                b'9 <- peek ptr9
+                b'9::Word8 <- peek ptr9
                 if b'9 < 128 then ok (val'8 .|. (fromIntegral b'9 `shiftL` 56)) 9 else do
                 let !val'9 = (val'8 .|. (fromIntegral (b'9 .&. 0x7F) `shiftL` 56))
                     !ptrA = ptr9 `plusPtr` 1
                 if ptrA >= end then more else do
 
-                b'A <- peek ptrA
+                b'A::Word8 <- peek ptrA
                 if b'A < 128 then ok (val'9 .|. (fromIntegral b'A `shiftL` 63)) 10 else do
                 err
 

--- a/stack-8.0.1.yaml
+++ b/stack-8.0.1.yaml
@@ -1,0 +1,19 @@
+packages:
+- '.'
+- descriptor/
+- hprotoc/
+resolver: nightly-2016-05-21
+compiler: ghc-8.0.1
+setup-info:
+    ghc:
+        macosx:
+            8.0.1:
+                url: https://downloads.haskell.org/~ghc/8.0.1/ghc-8.0.1-x86_64-apple-darwin.tar.xz
+                content-length: 118496652
+                sha1: e7a1c558a095afbc4fb514256be940ddb0f23a29
+extra-deps:
+# For ghc-8.0.1
+- containers-0.5.7.1
+- directory-1.2.6.3
+- transformers-0.5.2.0
+- unix-2.7.2.0

